### PR TITLE
[Enhancement] enable use meta scan in simple min/max agg query

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        }
+        // no limit
+        if (scanOperator.getLimit() != -1) {
+            return new ArrayList<>();
+        }
+        Map<String, PartitionColumnFilter> columnFilters = scanOperator.getColumnFilters();
+        // no filter
+        if (columnFilters != null && !columnFilters.isEmpty()) {
+            return new ArrayList<>();
+        }
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java:157]: check scan's predicate, not getColumnFilters

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java:157]: btw, metaScanNode doesn't support work on special tablet/partitions?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        // we can only apply this rule to the queries met all the following conditions:
+        // 1. query on DUPLICATE_KEY table
+        // 2. no group by key
+        // 3. no `having` condition or other filters
+        // 4. no limit
+        // 5. only contain MIN/MAX agg functions
+        // 6. all arguments to agg functions are primitive columns
+        // 7. no expr in arguments to agg functions
+        // 8. all agg columns have zonemap index
+        if (table.getKeysType() != KeysType.DUP_KEYS) {
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java:106]: fixed

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java (hunk 1 above)
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java]: > check scan's predicate

fixed

Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java]: > btw, metaScanNode doesn't support work on special tablet/partitions?

it supports…I made a mistake here before

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 72 / 73 (98.63%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java|68|69|98.55%|[150]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/MetaScanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/LogicalPlanPrinter.java|1|1|100.00%|[]|

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 73 / 74 (98.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java|69|70|98.57%|[151]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/MetaScanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/LogicalPlanPrinter.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 73 / 74 (98.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java|69|70|98.57%|[111]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/MetaScanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/LogicalPlanPrinter.java|1|1|100.00%|[]|

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**